src/presentation/pages/Note.vue

```diff
@@ -203,18 +197,33 @@ function transformNoteHierarchy(noteHierarchyObj: NoteHierarchy | null, currentN
   };
 }
 
-const verticalMenuItems = computed<VerticalMenuItem>(() => {
-  return transformNoteHierarchy(noteHierarchy.value, noteTitle.value);
+const verticalMenuItems = computed<VerticalMenuItem[]>(() => {
+  if (!noteHierarchy.value) {
+    return [];
+  }
+
+  return [transformNoteHierarchy(noteHierarchy.value, noteTitle.value)];
 });
 
 watch(
   () => props.id,
-  () => {
+  async () => {
     /** If new child note is created, refresh editor with empty data */
     if (props.id === null) {
       useHead({
         title: t('note.new'),
       });
+    } else {
+      // Load note settings first
+      await loadNoteSettings(props.id);
+
+      // Create load options based on note settings
+      const loadOptions: NoteLoadOptions = {
+        loadHierarchy: noteSettings.value?.showNoteHierarchy === true,
```

Comment:
you can just cast to boolean 
```suggestion
        loadHierarchy: !!noteSettings.value?.showNoteHierarchy,
```

Reply:
not required removed